hphp/hack/src/hh_codesynthesis/agentGraphGenerator.py

```diff
@@ -119,46 +135,82 @@ class AgentGraphGenerator:
     the user can do a dry run to evaluate the quality of actual graph.
     The user must specify which on_model is going to use."""
 
-    _raw_model = ""
-
-    def __init__(self) -> None:
+    def __init__(
+        self, agent_distribution: List[int], solving_context: AgentGraphClingoContext
+    ) -> None:
         self.infra_agents: List[int] = []
         self.product_agents: List[int] = []
         self.edges: List[Set] = []
+        self._raw_model = ""
+        self.agent_distribution = agent_distribution
+        self.solving_context = solving_context
 
-    def infra_agent_evaluate(self, agent_number: int) -> None:
+    def add_infra_agent(self, agent_number: int) -> None:
         self.infra_agents.append(agent_number)
 
-    def product_agent_evaluate(self, agent_number: int) -> None:
+    def add_product_agent(self, agent_number: int) -> None:
         self.product_agents.append(agent_number)
 
-    def edge_evaluate(self, left_agent: int, right_agent: int) -> None:
+    def add_edge(self, left_agent: int, right_agent: int) -> None:
         self.edges[left_agent].add(right_agent)
 
-    def evaluate(self, m: clingo.Model) -> None:
-        self.infra_agents = []
-        self.product_agents = []
+    def validate(
```

Comment:
This is great! Nit: I think having the validate script use assert instead of returning False might give better test signal if something breaks later

Reply:
assert with error message.